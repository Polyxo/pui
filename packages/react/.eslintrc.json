{
  "root": true,
  "parser": "@typescript-eslint/parser",
  "plugins": ["@typescript-eslint"],
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:storybook/recommended"
  ],
  "rules": {
    "no-unused-vars": "off",
    "no-console": "error",
    "@typescript-eslint/no-unused-vars": ["error"]
  }
}
