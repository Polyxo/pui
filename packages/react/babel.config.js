module.exports = {
  presets: [
    "@babel/preset-env",
    "@babel/preset-react",
    "@babel/preset-typescript",
  ],
  plugins: [
    "dev-expression",
    "@babel/plugin-syntax-dynamic-import",
    "@babel/plugin-syntax-import-meta",
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-export-namespace-from",
    "@babel/plugin-proposal-export-default-from",
  ],
};
