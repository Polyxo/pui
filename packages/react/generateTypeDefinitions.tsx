import { rm, stat, readdir, writeFile, mkdir } from 'fs/promises';
import path from 'path';
import { parse } from 'react-docgen-typescript';

const distUrl = path.resolve(__dirname, '../../website/types/');

const options = {
  savePropValueAsString: true,
  propFilter: (prop, component) => {
    if (prop.declarations !== undefined && prop.declarations.length > 0) {
      const hasPropAdditionalDescription = prop.declarations.find(
        (declaration) => {
          return !declaration.fileName.includes('node_modules');
        }
      );

      return Boolean(hasPropAdditionalDescription);
    }

    return true;
  },
};

export default async function walk(directory: string) {
  let fileList: string[] = [];

  const files = await readdir(directory);
  for (const file of files) {
    const p = path.join(directory, file);
    if ((await stat(p)).isDirectory()) {
      fileList = [...fileList, ...(await walk(p))];
    } else {
      fileList.push(p);
    }
  }

  const tsxFileList = fileList.filter(
    (file) => file.endsWith('.tsx') && !file.includes('stories')
  );

  await Promise.all(
    tsxFileList.map(async (file) => {
      const propTypes = JSON.parse(JSON.stringify(parse(file, options)));

      try {
        await mkdir(path.dirname(path.join(distUrl, file)), {
          recursive: true,
        });
      } catch (error) {
        console.log('collectTypes error', error);
      }

      console.log(
        'propTypes',
        path.join(distUrl, file.replace('.tsx', '.json'))
      );
      await writeFile(
        path.join(distUrl, file.replace('.tsx', '.json')),
        JSON.stringify(propTypes, null, 2)
      );
      return null;
    })
  );

  return fileList;
}

async function startGenerateTypes() {
  await rm(distUrl, { recursive: true, force: true });

  await walk('./src');
  process.exit();
}

startGenerateTypes();
