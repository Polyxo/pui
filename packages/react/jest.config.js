module.exports = {
  preset: "ts-jest",
  testEnvironment: "jsdom",
  transform: {
    "^.+\\.tsx?$": "ts-jest",
  },

  roots: ["<rootDir>/src"],
  testMatch: [
    //"**/__tests__/**/*.+(ts|tsx|js)",
    "**/?(*.)+(spec|test).+(ts|tsx|js)",
  ],
};
