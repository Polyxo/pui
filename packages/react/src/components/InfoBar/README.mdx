import { Meta, Story } from '@storybook/addon-docs/blocks';

### Usage

`InfoBar` will contain the default links.

Do not use it in other places.
