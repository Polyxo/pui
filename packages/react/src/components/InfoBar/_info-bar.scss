/*
@include '../../globals/scss/vars';
@import '../../globals/scss/colors';
@import '../../globals/scss/typography';
@import '../../globals/scss/spacing';
@import '../../globals/scss/helper-mixins';
@import '../../globals/scss/vendor/@un/import-once/scss/import-once';
@import '../../globals/scss/css--reset';
@import '../Form/form';

@include exports('info-bar') {
  .#{$prefix}--info-bar {
    background: var(--support-03);
    @include reset;
    display: flex;
    padding: 0.5rem 0;
    line-height: 22px;
  }
}
*/
