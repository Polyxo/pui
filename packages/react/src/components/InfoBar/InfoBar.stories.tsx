import React from "react";
import markdown from "./README.mdx";
import InfoBar from ".";
import Link from "../Link";

export default {
  title: "Components/UI Elements/InfoBar",
  component: InfoBar,
  parameters: {
    componentSubtitle: "Component",
    status: "released",
    mdx: markdown,
  },
};

export const InfoBarDefault = (args) => (
  <InfoBar {...args}>
    InfoBar with content. <Link href="#">Click here</Link> to do Something.
  </InfoBar>
);

const infobarsourcecode = `
  import { InfoBar, Link  } from "@progressiveui/react";

  <InfoBar>
    InfoBar with content. 
    <Link href="#">
        Click here
    </Link>
     to do Something.
  </InfoBar>
`;

InfoBarDefault.story = {
  parameters: {
    docs: {
      source: {
        code: infobarsourcecode,
      },
    },
  },
};
