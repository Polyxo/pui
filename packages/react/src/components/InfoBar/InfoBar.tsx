import * as React from 'react';
import type { PropsWithChildren } from 'react';
import classNames from 'classnames';
import Wrapper from '../Wrapper';
import useSettings from '../../hooks/useSettings';
import { ScreenSize } from '../../typesLegacyBB/utils';

/** The InfoBar shows very important information on top of the page. */
type InfoBarProps = PropsWithChildren<{
  pageWidth?: ScreenSize;
  wrapperClassName?: string;
  className?: string;
  id?: string;
}>;

const InfoBar: React.FC<InfoBarProps> = ({
  children,
  className,
  id,
  pageWidth,
}) => {
  const { prefix } = useSettings();

  const wrapperClasses = classNames(`${prefix}--info-bar`, className);

  return (
    <div id={id} className={wrapperClasses}>
      <Wrapper pageWidth={pageWidth}>{children}</Wrapper>
    </div>
  );
};

InfoBar.displayName = 'InfoBar';

export default InfoBar;
