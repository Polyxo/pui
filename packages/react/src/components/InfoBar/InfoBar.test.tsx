//import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import InfoBar from "./InfoBar";

describe("InfoBar Component", () => {
  // Test basic rendering
  it("renders the InfoBar with content", () => {
    render(<InfoBar>Important Information</InfoBar>);
    expect(screen.getByText("Important Information")).toBeInTheDocument();
  });

  // Test custom class name
  it("applies custom class names", () => {
    const customClass = "my-custom-class";
    render(<InfoBar className={customClass}>Important Information</InfoBar>);
    expect(screen.getByText("Important Information").parentNode).toHaveClass(
      customClass
    );
  });

  // Test id attribute
  it("sets id when provided", () => {
    const id = "info-bar-id";
    render(<InfoBar id={id}>Important Information</InfoBar>);
    expect(
      screen.getByText("Important Information").parentNode
    ).toHaveAttribute("id", id);
  });

  // Test different page widths
  it("renders with specified page width", () => {
    const pageWidth = "lg";
    render(<InfoBar pageWidth={pageWidth}>Important Information</InfoBar>);
    // Assuming the pageWidth prop affects the class of the Wrapper component
    expect(screen.getByText("Important Information")).toHaveClass(
      `wfp--wrapper--width-${pageWidth}`
    );
  });

  // Snapshot test
  it("matches snapshot", () => {
    const { asFragment } = render(<InfoBar>Snapshot InfoBar</InfoBar>);
    expect(asFragment()).toMatchSnapshot();
  });
});
