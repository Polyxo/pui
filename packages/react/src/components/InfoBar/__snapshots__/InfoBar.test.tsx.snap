// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBar Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="wfp--info-bar"
  >
    <div
      class="wfp--wrapper"
    >
      Snapshot InfoBar
    </div>
  </div>
</DocumentFragment>
`;
